pkg/console/operator/sync_v400.go

```diff
@@ -46,76 +46,85 @@ func (co *consoleOperator) sync_v400(updatedOperatorConfig *operatorv1.Console,
+	rtConditionType := "RouteSync"
+	co.HandleProgressingOrDegraded(updatedOperatorConfig, rtConditionType, rtErrReason, rtErr)
 	if rtErr != nil {
-		klog.V(4).Infof("incomplete sync: %v", fmt.Sprintf("%v: %s", "route", rtErr))
+		klog.V(4).Infof(fmt.Sprintf("%v %v: %s", rtConditionType, rtErrReason, rtErr))
 		return rtErr
 	}
 
-	svc, svcChanged, svcErr := co.SyncService(set.Operator)
+	svc, svcChanged, svcErrReason, svcErr := co.SyncService(set.Operator)
 	toUpdate = toUpdate || svcChanged
-	co.HandleProgressing(updatedOperatorConfig, "ConsoleServiceSync", svcErr)
+	svcConditionType := "ServiceSync"
+	co.HandleProgressingOrDegraded(updatedOperatorConfig, svcConditionType, svcErrReason, svcErr)
 	if svcErr != nil {
-		klog.V(4).Infof("incomplete sync: %v", fmt.Sprintf("%q: %v", "service", svcErr))
+		klog.V(4).Infof(fmt.Sprintf("%v %v: %s", svcConditionType, svcErrReason, svcErr))
 		return svcErr
 	}
 
-	cm, cmChanged, cmErr := co.SyncConfigMap(set.Operator, set.Console, set.Infrastructure, rt)
+	cm, cmChanged, cmErrReason, cmErr := co.SyncConfigMap(set.Operator, set.Console, set.Infrastructure, rt)
 	toUpdate = toUpdate || cmChanged
-	co.HandleProgressing(updatedOperatorConfig, "ConsoleConfigMapSync", cmErr)
+	cmConditionType := "ConfigMapSync"
+	co.HandleProgressingOrDegraded(updatedOperatorConfig, cmConditionType, cmErrReason, cmErr)
 	if cmErr != nil {
-		klog.V(4).Infof("incomplete sync: %v", fmt.Sprintf("%q: %v", "configmap", cmErr))
+		klog.V(4).Infof(fmt.Sprintf("%v %v: %s", svcConditionType, cmErrReason, cmErr))
 		return cmErr
 	}
 
-	serviceCAConfigMap, serviceCAConfigMapChanged, serviceCAConfigMapErr := co.SyncServiceCAConfigMap(set.Operator)
+	serviceCAConfigMap, serviceCAConfigMapChanged, serviceCAErrReason, serviceCAConfigMapErr := co.SyncServiceCAConfigMap(set.Operator)
 	toUpdate = toUpdate || serviceCAConfigMapChanged
-	co.HandleProgressing(updatedOperatorConfig, "ServiceCASync", serviceCAConfigMapErr)
+	serviceCAConditionType := "ServiceCASync"
+	co.HandleProgressingOrDegraded(updatedOperatorConfig, serviceCAConditionType, serviceCAErrReason, serviceCAConfigMapErr)
 	if serviceCAConfigMapErr != nil {
-		klog.V(4).Infof("incomplete sync: %v", fmt.Sprintf("%q: %v", "serviceCAconfigmap", serviceCAConfigMapErr))
+		klog.V(4).Infof(fmt.Sprintf("%v %v: %s", serviceCAConditionType, serviceCAErrReason, serviceCAConfigMapErr))
 		return serviceCAConfigMapErr
 	}
 
-	trustedCAConfigMap, trustedCAConfigMapChanged, trustedCAConfigMapErr := co.SyncTrustedCAConfigMap(set.Operator)
+	trustedCAConfigMap, trustedCAConfigMapChanged, trustedCAErrReason, trustedCAConfigMapErr := co.SyncTrustedCAConfigMap(set.Operator)
 	toUpdate = toUpdate || trustedCAConfigMapChanged
-	co.HandleProgressing(updatedOperatorConfig, "TrustedCASync", trustedCAConfigMapErr)
+	trustedCAConditionType := "ConfigMapSync"
+	co.HandleProgressing(updatedOperatorConfig, trustedCAConditionType, trustedCAErrReason, trustedCAConfigMapErr)
```

Comment:
Dont we wanted to switch to `HandleDegraded`, based on https://github.com/openshift/console-operator/pull/265#discussion_r317771052